tests/unit/core/test_Wallet.py

```diff
@@ -0,0 +1,136 @@
+from twisted.trial import unittest
+
+from twisted.internet import threads, defer
+from lbrynet.core.Wallet import Wallet
+
+test_metadata = {
+'license': 'NASA',
+'fee': {'USD': {'amount': 0.01, 'address': 'baBYSK7CqGSn5KrEmNmmQwAhBSFgo6v47z'}},
+'ver': '0.0.3',
+'description': 'test',
+'language': 'en',
+'author': 'test',
+'title': 'test',
+'sources': {
+    'lbry_sd_hash': '8655f713819344980a9a0d67b198344e2c462c90f813e86f0c63789ab0868031f25c54d0bb31af6658e997e2041806eb'},
+'nsfw': False,
+'content_type': 'video/mp4',
+'thumbnail': 'test'
+}
+
+
+class MocLbryumWallet(Wallet):
+    def __init__(self):
+        pass
+    def get_name_claims(self):
+        return threads.deferToThread(lambda: [])
+
+    def _save_name_metadata(self, name, claim_outpoint, sd_hash):
+        return defer.succeed(True)
+
+
+class WalletTest(unittest.TestCase):
+
+    def _check_exception(self, d):
```

Comment:
It looks like trial has built in functionality for this:
http://twistedmatrix.com/documents/current/api/twisted.trial._synctest._Assertions.html#failureResultOf